### ISSUE TITLE
[UX] Add new front-end theme (Basis) in core

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1361]
## Summary

We need a new front-end theme in core. Here is a start at a list of requirements based on the comments below:
* lightweight (not too many template / theme function overrides)
* attractive (like Squarespace or WP)
* responsive (yes, obviously)
* well-thought-out front end architecture (what does this mean?)
* support full-width image blocks
* allow for easy change of:
  * colors (not necessarily via color module)
  * ~~fonts~~
  * ~~corner styles~~

[See all existing contrib themes for backdrop, here.](http://dev-backdropthemes.pantheon.io) thanks @biolithic 

## Current Status

We have a new **feature branch [1.x-basis](https://github.com/backdrop/backdrop/tree/1.x-basis)**. All the latest code for Basis should be in that branch.

- [x] The main menu needs dropdown styling.
- [x] The main menu needs to be styled as a single-row of links that takes up 100% width.
- [x] The header CSS is still a bit wonky. The user menu links are huge.
- [x] We need a screenshot of the completed theme added as `screenshot.png`
- [x] **Not doable for 1.5** ~~The hero block in mockups went edge-to-edge and rested directly against the header block.~~ 
- [ ] @**Responsive tab JS isn't being added yet**
  Thinking we should move `core/themes/seven/js/script.js` out of `seven` and put it somewhere where `seven` and `basis` can both make use of it. We could make it a library?
- [x] **There's some PHP in `header.tpl.php` @wesruv would prefer to see in `template.php`, but was having issues**
  Stumped. I made `template_preprocess_header` did all of the work I needed and added an index to `$variables` to be printed as a class in `header.tpl.php`, but for some reason `$variables` did not show my update...
https://github.com/wesruv/backdrop/blob/cecabe3a47f46ff1c1aa0212cc553db7c94e14c1/core/themes/basis/templates/header.tpl.php#L26
- [x] Default config to make Basis default theme
- [x] Open Sans font isn't being added yet
- [x] There are a number of small todo's that need to be addressed, I plan on getting to them later today
- [x] All tests are passing.

***

## Original issue

This was mentioned in #1359:

@stephit, https://github.com/backdrop/backdrop-issues/issues/1359#issue-118069106
>> More/better default themes, along with theme creator engine. Bartik is great but it is very 2003. Come up with two or three really nice, lightweight, attractive and RESPONSIVE themes that come in Core so that small businesses can just pick something easily and customize it. Add a theme engine creator where you can change fonts, colors and corner styles. Along with layouts, this would be so helpful for small businesses!

@quicksketch, https://github.com/backdrop/backdrop-issues/issues/1359#issuecomment-158604336

> I thought for sure we already had an issue for making a new core theme but I can't find it. Absolutely agree. We started with the admin theme but we'd love a new front-end theme as well.

...but I'm sure that it must have been in almost everyone's head for some time. It would differentiate us from D7/D8 and honestly, it is about time that Bartik R.I.P.

May I suggest a few things:

1. Start a "contest" and call out for theme designers to propose their ideas/mockups here.
2. In the meantime, lets vote for [one of the already contributed/ported themes](https://backdropcms.org/themes) and build upon it.
3. Should we establish this as a thing we do for every major release like Ubuntu does with their wallpaper contest?

This is a and/or list and I would love to hear people's thoughts.

---
PR https://github.com/backdrop/backdrop/pull/1557

**Comment by sutibun** [comment]
1. > Start a "contest" and call out for theme designers to propose their ideas/mockups here.
    
    Like though wonder how many Bd theme designers there are at the moment. Maybe few not bad? The monkey sort of a designer, at least good enough for site. Maybe monkey can participate. Just have no exp with Bd/Drupal theming.
2. > In the meantime, lets vote for one of the already contributed/ported themes and build upon it.
    
    Following two Bd themes has monkey's vote:
    
    - [Colihaut](https://backdropcms.org/project/colihaut) Why? Because it's simple and can cover different types of sites. *More importantly*, you can customize the site's color from the theme's settings so won't look the same as others.
    - [Cleanish](https://backdropcms.org/project/cleanish) Why? Black and white are Bd's colors (based on current site at least) Plus, it's a simple theme fitting for any site.
    
    Another reason went with those two is because they're the only Bd themes that just worked when I installed them. The others I installed and activated but got nothing. (maybe it was the layouts?) Anyway, that's that.
3. > Should we establish this as a thing we do for every major release like Ubuntu does with their wallpaper contest?
    
    See 1.

**Comment by biolithic** [comment]
I generally port popular themes that other technologies may use well.  For
example, "I use UI framework X in Ruby on Rails/Wordpress/Django/etc".
 "Well, now you can use the same framework/knowledge in
a BackdropCMS"...ease of community entry.  So I don't do a lot of theme
designing, I just am working on making it easier for people to port stuff
or transition to Backdrop using existing knowledge.

I like the contest idea so far.  Here's a crazy idea -- what if a core
theme was a feminine type theme?  You have Seven, or a black and white one,
and a cute one with pretty pastels -- market it to GirlDevelopIt,
CodeWithCupcakes, BlackGirlsCode, CodeForAmerica, JewelBots, and that kind
of crowd.  That's my crazy idea of the day and probably my choice.




On Sat, Nov 21, 2015 at 1:42 PM, Steven Villaverde <notifications@github.com

**Comment by mikemccaffrey** [comment]
I definitely agree that this is a very important task for us. In the short term, we might benefit from backdrop starting off with the familiar (though outdated) drupal look and feel, but if we want to foster our own separate user base, we need our own tailored starting point for backdrop sites.

Would it perhaps be beneficial for us to list out the requirements that we think the new default theme should have? For example, I think supporting full-width image blocks is a must for any modern template. What do you folks think the new theme should look/feel/work like?

**Comment by sutibun** [comment]
> I generally port popular themes that other technologies may use well… I just am working on making it easier for people to port stuff or transition to Backdrop using existing knowledge.

Nice.

I think adding **too many** restrictions wouldn't be beneficial at this point. Maybe if we had a lot of designers around. I'd say we keep things loose. Something like:

- be mobile friendly
- the theme can fit different types of sites… not too niche (e.g. A dark gloomy look might fit a punk rock site but not a mommy site)

**Comment by biolithic** [comment]
Hi, I made a site showcasing all the current contrib themes for Backdrop: 
http://dev-backdropthemes.pantheon.io

They're not all done, might find some visual bugs on all of them, etc... (which we can fix).
I just uploaded and enabled them onto this site with demo content.  I hope it makes it easier for everyone to start make opinions about "more themes" or "core themes" or whatever.

If you look at the site, thanks!

**Comment by docwilmot** [comment]
Thats really nice @biolithic. I need to work on my Colihaut CSS it seems.

**Comment by biolithic** [comment]
Heh, there's "problems" with every theme.  This is just so people can make
more informed decisions about do we need more themes or should we improve
an existing one people like (I feel like half the themes I pulled off
Github because they are not on Backdropcms.com yet, so people may not know
about them.  Some are in early development and I can update the site every
weekend with what's changed).  Thanks for viewing!

**Comment by klonos** [comment]
> ...I can update the site every weekend with what's changed.

If we get drush working with Backdrop, than that would make your life easier. You could even automate this or we can set up some "official" showcase site that does that. In general :+1: and thanx.

**Comment by biolithic** [comment]
Hi, I've been working on some "front page" layouts.  Here is an example:

http://dev-backdropthemes.pantheon.io/zurb-5

How it works is:
-- use almost any theme
-- add the layout to a specific path
-- in a theme's settings, there are "add background image" settings you can
just add the background images through the UI.
-- go to the specific page and it will look like that

This is an experiment in overriding a theme's layout for a specific path.
The use case is: "our front page looks different than our content pages".
For example, Bartik has a nice "960 width".  But the front page "should
look more like the above link", so use a layout like this instead of
changes themes or writing CSS.

The images are just random images I chose -- not part of the theme or
layoiut.

*@sutibun <https://github.com/sutibun> thoughts?*

**Comment by wesruv** [comment]
RE: @mikemccaffrey:
> Would it perhaps be beneficial for us to list out the requirements that we think the new default theme should have?
... 
> What do you folks think the new theme should look/feel/work like?

* I think it should work with the color module, or have some way of easy customization so users that don't know CSS can have control over it's appearance
* I want it to look 'Squarespace good', undeniably slick. Methinks this is going to mean some imagery that comes with the default theme.
* I want to use modern layout techniques with fallbacks for older browsers, because I think it'd make us look cooler :stuck_out_tongue: For instance, @biolithic's link seems to be using vw to have the header be huge
* I'm starting to think about our Front End Architecture a lot more (class naming, markup patterns, coding guidelines, etc). We still have a lot of D7 front end architecture, which needs to go away and never come back. This could be a good opportunity to make up a good F.E.A. and see how it works for us.

**Comment by wesruv** [comment]
I'd also like to see our  base theme 'look like a site' out of the box, and not an empty listing.

We could put sample content for a homepage in a if statement block, if they've done stuff on the site don't show this garbage, if htey haven't, show sample content and a message telling them how to get started.

**Comment by quicksketch** [comment]
> I want it to look 'Squarespace good', undeniably slick. Methinks this is going to mean some imagery that comes with the default theme.

We could also go a long ways towards this goal by improving the standard profile, and implementing items like adding a front page layout (#1042). We can add better default content and/or site configuration as well.

**Comment by biolithic** [comment]
This is an interesting thread to read because people will say, "We need a
new theme"
...
but
...
want...vs...need

This actually could mean new layout, theme, imagery, dummy content,
documentation/demo, etc... have to read between the lines.

"To get more themers working on Backdrop, the new theme should use BEM and
libSASS and etc..."
vs
"To get more director of communications and VC's and marketing leads of
business/organizations working on Backdrop, all themes mandatory Color
module and more theme settings"
vs
"To get more diversity, we should avoid where we promote only that which
appeals to white male 'engineers'.  Backdrop is not currently used by 'the
enterprise' (whatever that is), so it should be more 'grass-roots-y'"
...etc...

https://github.com/backdrop/backdrop-issues/issues/1042 may be helped by
the presence of something like
https://backdropcms.org/project/juiced_up_layouts shown here:
http://dev-backdropthemes.pantheon.io/zurb-5

(The above link contains no custom CSS, only layout/theme settings applied
to a blank theme)

Will work on polishing up the existing themes and happy to help wherever I
can on this.  We couldn't really practice searching for, theming pages of,
paginating or project browsing lots of themes/layouts on Backdropcms.org
until we had some sample themes/layouts to throw on there, but now we do.

**Comment by mikemccaffrey** [comment]
> I'd also like to see our base theme 'look like a site' out of the box, and not an empty listing. We could put sample content for a homepage...

One option would be to pull in the README.md from the site root if there is no content on the site. Then we will be sure that there is a good introductory message that is seen and updated regularly.

**Comment by Graham-72** [comment]
Would it be possible to provide a wizard that leads a new user through some decisions about colour scheme, placement and style of navigation, page or blog type of site etc etc?

**Comment by wesruv** [comment]
@graham-72 that's what I'm thinking :)

**Comment by wesruv** [comment]
An awesome homepage that's calls to action happens to tell users how to customize their site

**Comment by docwilmot** [comment]
Re default front page content, there's this proposal:
https://github.com/backdrop/backdrop-issues/issues/519

I love the idea of a full-grown site on install. In https://github.com/backdrop/backdrop-issues/issues/519 we discussed a menu link for the default content. Would be nice if your site came with 
- a front page with its own layout
- a hello post
- and about us page
- contact page with its own layout https://github.com/backdrop/backdrop-issues/issues/1168#issuecomment-143649312 
- default banner image (or slider, but that'd be much)

And re a site wizard, there's [Setup Module](https://github.com/backdrop-contrib/setup)
Would be nice if the current installer ended with an option to "Use the BD Wizard to set up your site?", and that launched choices to enable the above options (default content, slider, contact, even users) and even then moves cleanly into Project Browser to enable modules, themes, and even "packages" to enable config-defined downloads that built entire sites - eg a one-click school site.

But this issue is about a new theme, so I'll stop rambling. 

Something like this: http://themes.muffingroup.com/cake/

**Comment by biolithic** [comment]
hey @klonos thanks for updating the list.  If we're going to work on something, we should know what we are working towards.  For example, I've never seen the Color module in production before.  Someone with more authority or money than me usually says, "I want this design - look at this Photoshop mockup/napkin. Now make it"...and I do or someone does (in some code).  However, maybe we are really missing an audience that would really use Color module (it's hard to do gradients/background graphics/shadows/etc with it).

I played around with some site-building this weekend trying to copy some websites and noting where themes/layouts have some issues.  We have a way to go but we can get there.

**Comment by mikemccaffrey** [comment]
I'd like to add a request to the list of requirements: We should have a theme that works well as a base theme when you are creating new custom themes. It has always struck me as weird that I've had to download a contrib theme to get a simple framework that I can build my own custom theme upon.

We could either do this by making the default theme simple enough to work as a base theme. Or we could provide both a fancy default theme and a simple base theme, perhaps even with the former being building upon the latter.

**Comment by Graham-72** [comment]
>We should have a theme that works well as a base theme when you are creating new custom themes

Yes please :+1:

**Comment by wesruv** [comment]
Here's an idea I had for the starter theme, and if you look at the recording from this weeks design meeting you can see my walk through and the feedback.
https://www.dropbox.com/s/xpkz5kwsvzgwqk4/starter-theme-idea.PSD?dl=0

![starter-theme-idea](https://cloud.githubusercontent.com/assets/5607236/12991545/9b589e8c-d0dd-11e5-84ec-f139643152b5.png)

**Comment by wesruv** [comment]
Feel free to take it, make it your own, code it, whatever.

After I make some effort I'll make my code available to for anyone that wants to fork/look at/make fun of that :stuck_out_tongue:

**Comment by wesruv** [comment]
Here's a quick idea for the design. Got a decent color scheme, and whipped up the background texture from scratch :smile: 

I'd love to hear thoughts, see other people's take, etc. I'm going to harass some designer friends for feedback as well. I feel like I could benefit from some brainstorming.

I think this is good enough for me to start coding, I'll hit this thread again when I've got some code to show.
![image](https://cloud.githubusercontent.com/assets/5607236/12996831/8fddafd8-d101-11e5-9838-632d1ffc7887.png)

![image](https://cloud.githubusercontent.com/assets/5607236/12996829/8a1fe638-d101-11e5-8b1b-f7446f777b2c.png)

**Comment by wesruv** [comment]
Oh, should mention... there's a cool kid way we could update that texture's color scheme via the color module and CSS. It'll mean that the texture will have to be a CSS bg image and not a content image, but I think that'd be a cool feature to have. 

This way we avoid any IP issues with photos or whatever and have something that is flexible enough for any color scheme.

**Comment by docwilmot** [comment]
Very very nice. I think though the checklist and the fly-in checklist thing are not going to be part of the theme, right? And the blocks with images will also need to be a new layout.

So to get this will require a bunch of new parts.

**Comment by wesruv** [comment]
@docwilmot you are correct, to accomplish this idea we'd have to make some other pieces and parts.

Trying to think of this more as a 'first run experience'. It's a little off topic from this ticket, but I'd like to think about all of this at once.

**Comment by klonos** [comment]
I like it a lot! :+1:

I like the idea of the background image being a configurable CSS generated pattern too, but don't know if it would be too much coding-wise. I believe that it would be more useful to focus on a views-generated slider (of file/image nodes) in the header instead.

**Comment by wesruv** [comment]
Responding @mikemccaffrey's base theme: I totally agree, when I was newer to Drupal I assumed I could use Bartik as a base theme and for the project I was on that really hurt us in the long run.

Right now I'm thinking that the base theme will be built so it's easy to take out the pieces you don't like. For example, abstracting the CSS into different files so you could omit the ones that don't work for you. If aggregation is turned on there's no downside to doing this as long as I'm not making an 'unreasonable' amount of CSS files.

I'd also like to take a note from D8 and put all of the core templates in our base theme. When I was new to Drupal I had a really hard time understanding where templates came from. It's a bit more work for us since it is potentially duplicated code, but I think it makes for a better DX. That will also lead us to provide a recommendation for how to organize a bunch of TPL files (see [D8's Stable theme's template folder](http://cgit.drupalcode.org/drupal/tree/core/themes/stable/templates?h=8.0.3)), which I also like.

**Comment by mikemccaffrey** [comment]
@wesruv I love the design! The only thing I would change is that the top-right of the header bar should probably be the user account links, rather than social media, since that is a more pressing need right out of the box.

Wait, did D8 move all of the templates out of core and put them in the base theme, or did it just make copies of everything there? Personally, I'm thinking it would be best to make the base theme as simple as possible, but include a copy of all of the core templates and css in a non-listed template called example. Does that make sense?

**Comment by Graham-72** [comment]
Great to see this. I can't wait to be using it. Thank you.

**Comment by wesruv** [comment]
@mikemccaffrey they made copies. `stable` and `classy` themes both have copies of every template from core. `classy` is based on `stable`, but it adds the markup/classes you're used to seeing from D7. 

They're saying this will give them the flexibility to update markup that's in core underneath the `stable` theme, so they can stay up to date, but the output of Drupal doesn't change/break sites.

I dig the example route :smile:, by this:
> but include a copy of all of the core templates and css in a non-listed **template** called example.

Do you mean this:
> but include a copy of all of the core templates and css in a non-listed **theme** called example.

If so I'm on the same page, if not I'm confused?

**Comment by wesruv** [comment]
Updated the template based on feedback from Marissa at Lullabot and your guys' feedback.

PSD is still at: https://www.dropbox.com/s/xpkz5kwsvzgwqk4/starter-theme-idea.PSD?dl=0

## Logged in
![image](https://cloud.githubusercontent.com/assets/5607236/13161460/d0878710-d66b-11e5-976d-9f0339ef49a1.png)

### Logged in, Get started pane open
![image](https://cloud.githubusercontent.com/assets/5607236/13161476/f415388a-d66b-11e5-8230-4e0e3502f910.png)

## Logged out
![image](https://cloud.githubusercontent.com/assets/5607236/13161484/00095f2c-d66c-11e5-8e1d-e1ecc1bf83cd.png)

### Logged Out, Login box open
![image](https://cloud.githubusercontent.com/assets/5607236/13161507/32c1e146-d66c-11e5-81ed-ffef74b9646c.png)

**Comment by wesruv** [comment]
I've **just** started, so there's barely anything there, but figured I'd share.

The repo I'm using (with config) can be found here:
https://github.com/wesruv/backdrop-starter-theme

Theme is here:
https://github.com/wesruv/backdrop-starter-theme/tree/drop-theme/themes/drop

**Comment by klonos** [comment]
Just :heart: Drop sitting playfully at on top of the footer!

**Comment by klonos** [comment]
...will definitely give it a spin over the weekend.

**Comment by wesruv** [comment]
To be clear, at it's best my repo looks like this right now:
![image](https://cloud.githubusercontent.com/assets/5607236/13167596/981cf68a-d6a2-11e5-984d-ac31ad9826fb.png)

Not much to test yet, just thought I'd share where I'll be working.

**Comment by klonos** [comment]
> To be clear, at it's best my repo looks like this right now:

I know. I gave it a go. It's good that you've got the code on GitHub though so we can watch for updates and also file issues against it. Any chance you can transfer it over to the official contrib repo?

**Comment by wesruv** [comment]
Hrmm... how about when it starts resembling a first draft I make it a separate repo and throw it in contrib as an alpha?

**Comment by docwilmot** [comment]
I think less confusing to develop as a branch on backdrop repo. That, and also the official theme shouldn't be available as contrib, I think.

**Comment by docwilmot** [comment]
I think as well @wesruv delivering this as shown will be difficult unless you do #698 first. The blocks and region styling in the mockup image cannot be accomplished in Backdrop with just the theme, so for example the grey background would need to be the background of a layout, which as you pointed out in #698 only makes sense if there is standard markup for "regions" in all/most layouts. Which means we need to deal with layout region markup first before this theme can style them.

Alternatively, build this with styles for the current core layouts, then restyle it again once we've settled on the layout markup standards. But that seems the wrong way around.

**Comment by wesruv** [comment]
Re: Repo location
I think it'd be easier for testing it as a base theme if it were in contrib for now, and I don't want to assume that this will be the core theme until we're sure.

I've heard an interesting case for having some core themes in contrib, since the core theme can't change in certain ways, contrib could be a way to work on the core theme in between major versions. Would probably give it a slightly different name, but they've been talking about doing it in D8, and I can see the logic in it.

Re: layout
On the design call we discussed having the multi-column content coming from a view. That would give some examples the user could mess around with, and we'd be showcasing a great feature. Right now I'm using the 1 column theme and that's been working fine.

At this point I'm just blocking things in and not worrying too much about how it 'should be'.

I'll be using this as a base theme for an upcoming side-project that we're building in Backdrop so I can see what my dog food tastes like :stuck_out_tongue:

**Comment by klonos** [comment]
> On the design call we discussed having the multi-column content coming from a view. That would give some examples the user could mess around with, and we'd be showcasing a great feature.

:+1: 

> ...so I can see what my dog food tastes like :stuck_out_tongue: 

:yum: :smile:

**Comment by docwilmot** [comment]
> multi-column content coming from a view

How? The views grid plugin. Interesting idea. Incidentally I had  [PR for porting Views Responsive Grid](https://github.com/backdrop/backdrop/pull/1014) which I forgot about till now. Maybe I should finish that off.

In all though, its sounding more and more like we need a separate META issue for the out-of-the-box experience in 1.4.0. 
- new theme
- new front page layout
- default content with menu items (a switch in install profile to disable this?)
- default views (with Views Responsive Grid features?)
- likely changes to the install profiles
- a 'getting started' wizard UI
- new default front page blocks (the social icon links, the text block, the login block)
- redesign the header block?
- etc?

**Comment by wesruv** [comment]
@docwilmot good point. I made a parent ticket for these first run & site builder improvements:
https://github.com/backdrop/backdrop-issues/issues/1663

**Comment by wesruv** [comment]
Was thinking of calling the new default theme `drop`, but I found a problem with that ideal, lol :laughing: 
![image](https://cloud.githubusercontent.com/assets/5607236/13382411/e9e476c2-de41-11e5-82ff-1c5f17bb8170.png)

**Comment by klonos** [comment]
I don't get it. What's funny about it??? :confused:

**Comment by serundeputy** [comment]
https://xkcd.com/327/

**Comment by klonos** [comment]
:laughing: yeah, I get it now. :+1:

**Comment by wesruv** [comment]
I've been struggling with how much deviation from core is allowed as I've been working on this.

If I was following my whims (and completely ignoring what others might want/need) I'd probably end up overriding every theme function and template to change the CSS class naming convention, pare down markup, and I'd remove a lot of the base styles that come from core. (I have a very D8 `stable` theme mindset, `stark` theme was never `stark` enough for my tastes.)

The approach I'm currently using is a more 'moderate' approach. Some overrides where I feel techniques or markup **really** need an update. So far I've overridden a few TPL's, but nothing major. I've even added markup to some templates to make certain features work well (gasp!)

Then there would be the conservative approach, not overriding **any** TPL or theme function, and only using what comes out of core.

By Backdrop 2.0, I want to have something that's in-between my 'moderate' and 'whims' approach, and I'd argue that that will be best for Backdrop's core, contrib and theme devs. BUT that's a conversation and decision for a later time.

For this theme, inside of the 1.x scope, what amount of lee-way do you think should be taken with this theme? 

I'm particularly interested in the PMC's thoughts, since you have a final say.
@jenlampton @quicksketch @mikemccaffrey

Can quite a few liberties be taken (more than I'm taking now)?
Should the approach be more conservative than I'm currently being?
Crossing my fingers that the approach shouldn't be hard line conservative ... but should I be?

**Comment by biolithic** [comment]
Thanks for the reply and effort Wes.  Here are some of the original items
from above:


   - lightweight (not too many template / theme function overrides)
   - attractive (like Squarespace or WP)
   - small businesses can just pick something easily and customize it

So these types of comments probably disallow a new Stark, Zen, Aurora, or
otherwise strict base theme approach for all core themes.  In the case that
yours becomes a highly customized version and then we make Bartik to fit
the conservative approach, that I cannot say.  For example, is
http://qatar.sfs.georgetown.edu/ GeorgeTown University in Qatar built in
Drupal and did they have a dedicated themer concerned about markup?
Probably.  Will they switch to Backdrop in the next few months?  I would
bet not.  I guess a question to ask would be are we building the theme for
the Jason Pamental's and MortenDK's who cares about markup to use or are we
building for the Wordpress/Squarespace consumer who may not as much.

Just some thoughts I had this morning.  Keep in mind that no one's theme is
going to be used by everyone.  Is Bartik a nice theme?  Sure.  Did I use it
on D7?  No. Are the majority of websites adhering to a particular style or
framework using Bootstrap?  I would say yes.  Do you use it?  It's ok to
say no, but that doesn't stop people from doing a Google search for
"Bootstrap theme" or whatever is the most searched keywords for theme.

**Comment by wesruv** [comment]
For now I'm changing the name to `basis`, and I think it's good enough for an alpha, so now it's in contrib, w000t!

README has caveats and what have yous.

Let me know what you all think!
https://github.com/backdrop-contrib/basis

**Comment by jenlampton** [comment]
Officially pushing this to 1.5

**Comment by quicksketch** [comment]
I just wanted to post this great discussion Jen Simmons had the Drupal.org issue queue: https://www.drupal.org/node/2759849#comment-11546207 

It outlines what the purpose of a core theme should be (and not be) The bullet points:

- To be the theme that's used while a site is being built. 
- Teaching. / To give Drupal a look while people are learning about Drupal.
- To be an example of how to make a theme — for beginners.
- To ship — as the default core theme.

But really the entire post needs to be read to understand the reasoning on each of these.

**Comment by wesruv** [comment]
I'm apparently very opinionated on this topic, lol. Apologies for length.

> 1) To be the theme that's used while a site is being built.

I agree and I think Basis is doing well on this score. A little hard to know until it sees wider use.

> 2) Teaching. / To give Drupal a look while people are learning about Drupal.

Feeling great about Basis for this goal :smile: I can't say I considered this, but I think it worked out. We may consider beefing up the contrast on a few things, or at least having a higher contrast mode. Some of the subtlety of `table` styling could get lost on a blown out projector, and unfortunately there's a lot of `table`s in Backdrop admin/site builder experience still.

> 3) To be an example of how to make a theme — for beginners.

As someone that tried to use Bartik in this way, it failed _miserably_ for me. There were so many bugs and technical debt. 

I'm not sure if the intention was to use it as a base theme, but nothing told me not to. It seemed wise to base my work off of something that was in core, something that I believed to be under the highest scrutiny, and therefore, (in my mind) must be quality. I got soooo much technical debt on that project, and it went from a 'temporary theme' (where I thought I could mess around a little) to something that's lasted 6 years. The people still developing on that product have not forgiven me.

I felt like I learned what not to do by using Bartik.

&lt;/rant>

Anyway, I'm not sure how we're doing on this, I think this is going to be in the eyes of the community, I'm too close to it. 

I'm trying to walk a line of good for beginners to Backdrop, but solid for pros as well.

I want to assume that if the user wants to override the CSS that they're good enough to be dangerous in CSS, but assume that they might have little to no knowledge about working with CMSes, or Drupal/Backdrop specifically.

Things that are important to me on Basis from a dev standpoint:
* Low CSS specificity
* Trying to build for flexibility, writing features expecting more content than I intended for the space, that anything could be changed/tinkered with, all that fun stuff
* Making it easy for single bits to be replaced - e.g. splitting up the CSS files, someone could override just one component's styles very easily.

An example of walking the line between what's good for 'pros' and 'beginners':
She mentions BEM in context of what pros do, I want to get BEM into Backdrop core (>= 2.0.0). Not because I think that sites that use Backdrop need to follow that convention (I don't); but because it'll make for really low specificity, which is great to build on top of.

Beginners to CSS or Backdrop may think our class names look goofy, but it won't prohibit them from using/overriding those class names or styles, and the benefit of having absurdly low specificity from the platform your building on could be *huge*.

Using Bartik I had *many* specificity wars with it's CSS, and some of the worst offenders were only in the IE only stylesheet, so IE users would find loverly bugs like a yellow footer (my style) having white text on it (white was an `!important` style that only showed in the IE stylesheet).

> 4) To ship — as the default core theme.

Extremely thankful that we have a PMC, I'm not worried about this.

> On Branding

I mostly disagree with Jen here. She sees inherent conflict, and I think this section is disagreeing with some things that she points out earlier in this writing.

I think the core theme should look "Backdroppy". As she points out earlier, it will be the default theme that people will see in talks, tutorials, articles, etc. This is a big part of how Backdrop will be perceived, and it cuts two ways:

1. I don't want people to have an idea that Backdrop is like getting a lot of tools and lumber, with no instructions on how to build the house. I think giving the user something completely bare would come across that way.

2. I also don't want a first time Backdrop site builder to be cursing my name as they try to rip out a bunch of razmatazz that I put in just to make the site 'look nice' on first run.

I don't see these as being in un-resolvable conflict (at least not nowadays, at the launch of Drupal 7... much harder to do IMO).

I think if we can make it look 'on brand' but not 'branded', I think it's a win. It should be *as easy as possible* to rebrand, this is why I'm interested in things like allowing the user to change colors and styles from the UI (something Color module-esque, but better).

We need the Backdrop logo in the top left, but I added Drop to the footer on top of the "Powered by Backdrop" block. It'll 'look like Backdrop' in demos, but won't look like spin-off of BackdropCMS.org so a user has to untangle a bunch of crap to make it look like their own.

One thing her line of thought made me think of:
On my first Drupal site I saw the giant, ugly, bandit looking drop logo, all up in *my* header. First thing I tried to do was get rid of that, replace it with my own branding. Figuring that out *was not intuitive at all*.

While we don't have a default logo, how intuitive is it to change the site name, remove the Powered by Backdrop block, those kinds of things.

I'm not saying we should get rid of these elements, but there might be a great opportunity to make it more intuitive to change for a first time Backdrop user.

> Some people think it's important to have some kind of strong visual impact after the installation is complete — despite the fact the site is empty of content.

I'm one of these people, but it shouldn't be at the expense of the user's time.

Maybe in the install process we could have two options, new to Backdrop, populate a few things to show me some features (like the homepage & get started stuff I mocked up), and the "seasoned veteran", don't give me any of that crap.

OR maybe there's a button on the homepage that removes all of that config.

Not a perfect example, because you shouldn't build on it, but I learned A TON from the Commerce Kickstart distribution. I think there's value from being able to pick around something that's already built to see how things work under the hood.

> Some people even want to add content / add more content types because they worry about this first impression moment. I also disagree with this.

Agree that more content types isn't wise. However I think having a few custom block types (e.g. hero and card) could be used to make some nice structured custom content in blocks and will help us make a glossy looking initial install homepage.

> It seems almost every Drupal project has the budget for a custom theme (even if it's a tiny budget). If it doesn't, people don't use Drupal.

Think this should be changed ^ I don't see a reason why we couldn't make it so a distribution (if not core) could create fairly unique looking 5 page sites for people who don't have a theme budget. I'm not saying we get super opinionated, but I think a few small wise opinions can go a long way.

If we didn't want to have any opinion on appearance, we could go headless and be a content store. I've worked on those things, and I loathe them, I don't get a real web site for months.

If the Backdrop first run experience were a person, I imagine they'd talk quietly, have a few solid opinions, a nice clean looking outfit, and some comfy shoes.

**Comment by dyrer** [comment]
Hello, and sorry for my intervention 
In my opinion no need for a standard theme for Backdrop, this can change every year or what ever need to be done. WordPress changes default theme almost every year, why not and Backdrop.

**Comment by jenlampton** [comment]
Thanks for weighing in @dyrer :)  If we get enough people/interest to improve and iterate on core themes every year I'll be thrilled! For now, we have one (bartik) and need another (basis) but we are certainly not limiting our future. We're just trying to do the most with what we have right now :)

**Comment by jenlampton** [comment]
@wesruv I agree with your perspective on all points above, but did want to address one thing:

> how intuitive is it to change the site name, remove the Powered by Backdrop block, those kinds of things.

We've actually improved the situation in Backdrop a lot by separating these things from the theme. Logo / site name / etc are all block settings for the header block. There's a contextual link for that even if you don't really grock layouts yet. Removing the powered-by block is also as easy as removing the block from the layout. I'm curious to see if we've done a good enough job of simplifying this task. If not we can revisit, but we have already addressed the problem.

**Comment by quicksketch** [comment]
Thanks @wesruv for the feedback. Good points all around. My interpretation of the "branded" default theme was that the default Drupal core theme has frequently ended up looking very similar to Drupal.org itself. This has been true for every core theme back to Blue Marine. Sometimes it happened in reverse, for example Bartik came first but then drupal.org ended up looking like it two years later. I think everyone is in agreement (Jen Simmons too) that we want differentiation between the product's default theme and the product's official branding. The powered by block adjustment is great IMO because it's easy to remove and keeps our branding in a single easy location. The rest of Basis doesn't look anything like BackdropCMS.org or our overall branding (though it does still have the "Backdroppy" feel to it, because it's a default Backdrop site after all).

I've been posting a bunch of feedback to the Basis issue queue today and a few PRs. Let's keep hammering on Basis and the necessary core changes to support it. I'm feeling good about our direction here.

**Comment by klonos** [comment]
Just wanted to note that in one of our conversations with @dyrer it was well pointed as a thing that strikes bad to someone coming from the WP-land is that we don't have many contrib themes yet. That is true. We only have less than a couple of dozen ones when the contrib projects count a few hundred. I do not have a solution for this, just wanted to point that fact out as something that we need to be aware of.

**Comment by klonos** [comment]
> Let's keep hammering on Basis and the necessary core changes to support it. I'm feeling good about our direction here.

👍

**Comment by Graham-72** [comment]
I am hoping, both because of its name and because Backdrop needs this , that Basis will be not just a good theme in for people getting started but also an enduring platform for those of us who like to develop customised subthemes. I used to use Zen for this purpose in D7. I have used Bartik in Backdrop, and am now trying to use Basis, and am finding it not altogether easy to customise a header that is correctly responsive and works with different devices and browsers.

This is probably my fault for not yet understanding what can and cannot be done, nor how to do it 'properly' with Bootstrap. I am going to persevere because the aim is so important for future work. Maybe we need a 'Basis User Group' to share experiences and do's and don'ts?

**Comment by wesruv** [comment]
> I am hoping, both because of its name and because Backdrop needs this , that Basis will be not just a good theme in for people getting started but also an enduring platform for those of us who like to develop customised subthemes.

*Definitely* my intention.

I'm sorry to hear that about the header, but that makes a lot of sense now that you've mentioned it! 

I'll have to rethink some of the decisions I made. I thinking my header styling is too opinionated and is probably getting in your way if you want to do something different with it.

I've seen one of your sites where the header was tall and had a bg image, are there other sites you've worked on? It'd be great for me to see some implementations.

**Comment by wesruv** [comment]
![image](https://cloud.githubusercontent.com/assets/5607236/18188316/35d41a54-707f-11e6-8bcb-a81c5376cd40.png)

Here's a thought

**Comment by Graham-72** [comment]
@wesruv :+1:  for your latest thought
> are there other sites you've worked on? It'd be great for me to see some implementations.

Here are a couple of my sites that I'm working on
![welcome to the jars of clay ceramics painting studio and cafe jars of clay](https://cloud.githubusercontent.com/assets/10559274/18194507/04cd0f10-70de-11e6-9d1d-9afb6a6adb4d.png)

In this one (Jars of Clay) the image at top right on the home page is a views slideshow block. On other pages it is a single image chosen to relate to the content of the page i.e. different image on each page. On a smartphone the image is almost the full screen width.

For a different site I just want centered title, slogan and primary menu like this (work in progress)
![welcome catherine larner](https://cloud.githubusercontent.com/assets/10559274/18194602/b65fdf00-70de-11e6-8194-4b108326135b.png)

**Comment by jenlampton** [comment]
Do we have a working PR for this? If so I'd like to test it.  Can you please post a link to it in the bottom of the top post in this thread, and add appropriate issue tags? Reminder: we are one week away from release!

**Comment by jenlampton** [comment]
As mentioned in https://github.com/backdrop/backdrop-issues/issues/1699...

> basis is overriding the Admin bar css?

We need to be sure that basis doesn't override very much core CSS (and duplicates none), since every change we make to core CSS will become twice as much work if  that CSS is also in Basis. I'd be happy to do a review if we can get a PR ready for testing :)

**Comment by wesruv** [comment]
> We need to be sure that basis doesn't override very much core CSS (and duplicates none)

Yep, I'll try to remove all of that in the PR for putting Basis in core. Probably going to get a PR for contextual when I remove that code.

**Comment by jenlampton** [comment]
Woot, ty!

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by wesruv** [comment]
PR here: https://github.com/backdrop/backdrop/pull/1536

Thinking we may want a feature branch to merge to before we get it into 1.x, I'll leave that up to core committers to decide :stuck_out_tongue:

**Comment by jenlampton** [comment]
dibbs!!

**Comment by quicksketch** [comment]
> PR here: backdrop/backdrop#1536
> Thinking we may want a feature branch to merge to before we get it into 1.x, I'll leave that up to core committers to decide :stuck_out_tongue:

Great idea. I've fixed up that PR to get all tests passing and then merged into a new `1.x-basis` branch. This also includes the drop down support from https://github.com/backdrop/backdrop-issues/issues/2107, which aren't in 1.x yet because they need tests. We can add the styling for the dropdowns in the Basis branch in the mean time.

The current branch looks pretty good, but @wesruv called out some items that haven't been addressed. I can add a few things to that as well. I'm going to update the issue summary with the current status of things and we can use it has a checklist.

**Comment by quicksketch** [comment]
Here is the current out-of-box appearance with the `1.x-basis` branch. Several items need work yet. Annotated screenshot:

![selection_184](https://cloud.githubusercontent.com/assets/100206/18463144/f3da6790-793c-11e6-89e9-da2a837ac04e.png)

@wesruv Could you take a stab at the above tasks and file a PR against the new `1.x-basis` branch?

**Comment by klonos** [comment]
Which one is "the mockup"? Is it https://github.com/backdrop/backdrop-issues/issues/1361#issuecomment-244248246 ??

...also, it says on the task list that we should make this the default theme. Only on new installs. Right?

**Comment by docwilmot** [comment]
> In mockups, hero went edge-to-edge

I'd like to see that too. Would be good if layouts had a hero *region* without the container DIV.

**Comment by wesruv** [comment]
@klonos yes, it's only on new installs. We updated the "Standard" install profile, so existing sites won't be affected.

And yes, https://github.com/backdrop/backdrop-issues/issues/1361#issuecomment-244248246 is the latest mockup.

Re: full bleed hero: @quicksketch @docwilmot 
The issue I'm trying to wrap my head around is how to pop out of the layout without causing a bugs/horizontal scrollbars. Unfortunately viewport width does counts the width taken up by scrollbars as part of the viewport.

Meaning on Windows (and possibly other OS' if the scroll bar doesn't hide/show) a 100vw element will create a horizontal scroll bar :frowning: 

Unfortunately I don't think this is considered a bug.

I tried to get crazy with something like negative margin of `calc((100vw - 100%) / -2)`, but that didn't avoid the horizontal scroll bar :disappointed:

Full width backgrounds are possible (like the header bg color), but they need to be done on the region, not the block.

If we had the ability to add a utility class to a region, we could make this happen, but I don't think this will get solved by 1.5

Working on the other issues, should have a PR soon.

**Comment by wesruv** [comment]
@quicksketch not seeing contextual links on heroes... not sure what's up with that?

**Comment by jenlampton** [comment]
> The issue I'm trying to wrap my head around is how to pop out of the layout without causing a bugs/horizontal scrollbars. 

I think bootstrap has a way to do this, something like .jumbotron. Have you tried that @wesruv ?

**Comment by wesruv** [comment]
@quicksketch @jenlampton 
Questions about the dropdown menu:
* responsive menus JS is adding a title attribute to every single link with a value of "Description of LINKTITLE", it's really obnoxious and unhelpful
* I don't understand what these are: `.menu-dropdown a.current` `.menu-dropdown a.disabled`, I think we should document what these state classes are in the CSS.

Here's a PR for my updates:
https://github.com/backdrop/backdrop/pull/1554

**Comment by wesruv** [comment]
Looked through the JS, don't really understand when/how `.current` or `.disabled` get applied :/

Looks like the only reason disabled would get applied is if we add it? Current _looks_ like it's applied if you're on the page that's in the `href` attribute, but it doesn't seem to be doing anything?

**Comment by quicksketch** [comment]
> Full width backgrounds are possible (like the header bg color), but they need to be done on the region, not the block.
> If we had the ability to add a utility class to a region, we could make this happen, but I don't think this will get solved by 1.5

True, yeah no attributes/classes on regions. We've been fighting that but it may be a losing battle. I think the solution you've applied that works with what we have is a great start, even if it's not exactly what we wanted.

> I don't understand what these are: `.menu-dropdown a.current` `.menu-dropdown a.disabled`, I think we should document what these state classes are in the CSS.

The CSS that we have in `menu-dropdown.theme.css` was pulled directly from SmartMenu's own styles, but then tweaked to avoid any SmartMenus-specific markup. It looks like .disabled isn't used by default but it's supported if you manually add it: http://www.smartmenus.org/docs/#menu-item-classes, perhaps that's for things like mega menus or images that may get placed in menus that aren't actually links? But the docs say this only applies to the `<a>` tag... So perhaps it's for a scenario that we're not encountering.

It looks like `current` is applied if you enable an option for highlighting the active menu item (`markCurrentItem: true`), but this option is disabled by default and not possible to enable in our UI. Perhaps we should remove it from our CSS? We already have `.active` and `.active-trail` for this purpose. Though I must admit the prospect of moving this logic server-side instead of regenerating new menu markup on every page is awfully tempting. Perhaps a 2.x endeavor.

Great job updating the code @wesruv! I didn't run across anything that needed revision. So I've merged everything into the 1.x-basis branch for further testing.

**Comment by quicksketch** [comment]
Just trying to break things... can we make it so that Drop only shows up if the powered by block is the first block in the footer? Otherwise things like this happen:

![screenshot from 2016-09-14 02-02-01](https://cloud.githubusercontent.com/assets/100206/18506394/c7421dd8-7a20-11e6-89c8-50a9cc11872a.png)

**Comment by docwilmot** [comment]
> Full width backgrounds ... need to be done on the region, not the block.
We've been fighting that but it may be a losing battle. 

Thats sad that we cant do this with current core. I am going to push a contrib version of all the core layouts with the `.container` div *inside* the region divs

```
  <div class="l-wrapper">

      <?php if ($messages): ?>
        <div class="l-messages" role="status" aria-label="<?php print t('Status messages'); ?>">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $messages; ?>
          </div>
        </div>
      <?php endif; ?>

      <div class="l-page-title">
        <div class="l-wrapper-inner container container-fluid">
          <a id="main-content"></a>
          <?php print render($title_prefix); ?>
          <?php if ($title): ?>
            <h1 class="page-title"><?php print $title; ?></h1>
          <?php endif; ?>
          <?php print render($title_suffix); ?>
        </div>
      </div>

      <?php if ($tabs): ?>
        <nav class="tabs" role="tablist" aria-label="<?php print t('Admin content navigation tabs.'); ?>">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $tabs; ?>
          </div>
        </nav>
      <?php endif; ?>

        <div class="l-wrapper-inner container container-fluid">
          <?php print $action_links; ?>
        </div>

      <?php if (!empty($content['top'])): ?>
        <div class="l-top">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $content['top']; ?>
          </div>
        </div>
      <?php endif; ?>

      <div class="l-content" role="main" aria-label="<?php print t('Main content'); ?>">
        <div class="l-wrapper-inner container container-fluid">
          <?php print $content['content']; ?>
        </div>
      </div>

      <?php if (!empty($content['bottom'])): ?>
        <div class="l-bottom">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $content['bottom']; ?>
          </div>
        </div>
      <?php endif; ?>

  </div><!-- /.l-wrapper -->
```

So then `.l-top .container {width:100%}` can do a proper hero.

**Comment by jenlampton** [comment]
But basis is a theme. Override 10 core layout files, replace .container
with .container-fluid. problem solved, no?

On Sep 14, 2016 5:00 AM, "docwilmot" <notifications@github.com> wrote:

Full width backgrounds ... need to be done on the region, not the block.

We've been fighting that but it may be a losing battle.

Thats sad that we cant do this with current core. I am going to push a
contrib version of all the core layouts with the .container div *inside*
the region divs

  <div class="l-wrapper">

      <?php if ($messages): ?>
        <div class="l-messages" role="status" aria-label="<?php print
t('Status messages'); ?>">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $messages; ?>
          </div>
        </div>
      <?php endif; ?>

      <div class="l-page-title">
        <div class="l-wrapper-inner container container-fluid">
          <a id="main-content"></a>
          <?php print render($title_prefix); ?>
          <?php if ($title): ?>
            <h1 class="page-title"><?php print $title; ?></h1>
          <?php endif; ?>
          <?php print render($title_suffix); ?>
        </div>
      </div>

      <?php if ($tabs): ?>
        <nav class="tabs" role="tablist" aria-label="<?php print
t('Admin content navigation tabs.'); ?>">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $tabs; ?>
          </div>
        </nav>
      <?php endif; ?>

        <div class="l-wrapper-inner container container-fluid">
          <?php print $action_links; ?>
        </div>

      <?php if (!empty($content['top'])): ?>
        <div class="l-top">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $content['top']; ?>
          </div>
        </div>
      <?php endif; ?>

      <div class="l-content" role="main" aria-label="<?php print
t('Main content'); ?>">
        <div class="l-wrapper-inner container container-fluid">
          <?php print $content['content']; ?>
        </div>
      </div>

      <?php if (!empty($content['bottom'])): ?>
        <div class="l-bottom">
          <div class="l-wrapper-inner container container-fluid">
            <?php print $content['bottom']; ?>
          </div>
        </div>
      <?php endif; ?>

  </div>

So then .l-top .container {width:100%} can do a proper hero.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/backdrop/backdrop-issues/issues/1361#issuecomment-246990860>,
or mute the thread

**Comment by docwilmot** [comment]
> But basis is a theme. Override 10 core layout files, replace .container with .container-fluid. problem solved, no?

Only if you wanted every region full-width. Every region is wrapped in that `.container`, so then everything becomes full-width, and you'd then need to override the region CSS for each other region to make it fixed width again. Having to override every layout and every region just to get a hero (which core promises you can do) goes against my principle that it should just be doable from the UI as far as possible.

I guess instead we could expect that themers would build their themes anticipating that people might override the layout `.container`, and override the region CSS in the theme to make some regions fixed width. But it wouldnt work for Basis. 

So if I wanted to ensure that users could "easily" build a site with a full-width hero and fixed-width other regions, I could build a Basis-like theme that did the above (container-fluid, and fixed-width CSS all other regions), or build layouts that did just this. Building layouts eliminates the need to override anything.

p.s. the lengthy response is me trying to work this out in my head to see if I'm thinking straight. I think I am.

**Comment by wesruv** [comment]
I vote we work on making config for regions work in 1.6, and then we can have this feature in a nice sane way.

We _could_ override in the theme, and I _could_ use vw and bring on some bugs, and there are probably other solutions, but this late in the game, and for core especially, I think we go with what we have.

**Comment by docwilmot** [comment]
Completely agreed, which is why I'm doing this in contrib. We have enough to worry about in the next 24hrs. 😄 

But at least it will be available in contrib as an option from 1.5.0.

**Comment by wesruv** [comment]
@jenlampton I don't know if you saw in gitter, I'm not seeing contextual links on hero blocks, not sure what would cause that. 

On my phone, can't add that to the checklist at the top. #frakkingithub

**Comment by jenlampton** [comment]
Settings for regions still isn't an majority use case imo, and not a good
candidate for core. If we're doing it only because we have a single parent
.container class in the wrong place, it would be better to fix that root
problem than add an edge-casey feature. (Especially one that clutters the
UI.)

I thought part of the reason we added all these extra div's in the new set
of core themes was to solve this problem? so a theme could set width as
being either 100% or otherwise. If a theme wants a hero to be full-with it
should also be able to set fluid in that region and not-fluid everywhere
else. This is a solved problem in bootstrap, we must not have tested it
thoroughly when the layouts went in :(

...also I'm saying all this not looking at the code. :)

Not sure what the solution is for 1.5, maybe just no full-bleed in Basis?

**Comment by klonos** [comment]
I also say go with what we currently have and consider full width hero (or any region) in a separate issue. If we title that issue "No full-bleed in Basis", then we can consider it a bug (or known issue) of the initial mockup and target that issue for the 1.5.x cycle. If this is considered a feature ("Allow setting the hero block to be full width"), then we target it for 1.6.0 instead.

**Comment by quicksketch** [comment]
Another minor issue: the contextual menu element is invisible on the blue backgrounds. Perhaps add some kind of white wrapper around such elements to make them universally visible on hover?

![selection_186](https://cloud.githubusercontent.com/assets/100206/18537262/30f190a4-7ab9-11e6-8aeb-4fa1a06d1338.png)

**Comment by quicksketch** [comment]
The menu dropdowns have been merged in #2107, so this branch is officially unblocked. We have a number of small items to fix, but I think we should get this in and address the remaining items in follow-ups.

The `1.x-basis` branch is now up-to-date with the latest `1.x` branch. If tests are passing I'm going to merge this in so we at have tomorrow to work out the last issues with everyone on the same page.

**Comment by quicksketch** [comment]
Full PR: https://github.com/backdrop/backdrop/pull/1557

**Comment by docwilmot** [comment]
First off, Basis is fantastic. Well done.

But I for some reason thought we had default text instead of the "No content created yet" stuff. We need that for 1.6.0 if not.

**Comment by jenlampton** [comment]
I'm also feeling like with the shiny new Basis it would be fab to get some default content. Hopefully it's not very tricky to get that added. Since there is someone (thank you @docwilmot !!!) working on it now I've put it back in the 1.5 milestone, let's see how it goes :)

**Comment by wesruv** [comment]
Added PR for #2189 and #2188, #2190 is getting punted.

Here is a PR for some general fixes:
https://github.com/backdrop/backdrop/pull/1562

**Comment by quicksketch** [comment]
The `1.x-basis` branch has now been merged into `1.x` https://github.com/backdrop/backdrop/commit/3374874951a73f18635a1d082e5dd0773f5fa701.

Great job everyone and of course especially @wesruv!!! He who boldly not only developed the entire theme but also designed it and handled dozens of reviews over the last several months. Thank you!

**Comment by wesruv** [comment]
w0000000t!
